terminal.go

```diff
@@ -489,6 +472,123 @@ func (g *LightningTerminal) Run() error {
+// subServerMgr.
+func (g *LightningTerminal) initSubServers() {
+	// Overwrite the loop and pool daemon's user agent name so it sends
+	// "litd" instead of "loopd" and "poold" respectively.
+	loop.AgentName = "litd"
+	pool.SetAgentName("litd")
+
+	faradayServer := frdrpcserver.NewRPCServer(g.cfg.faradayRpcConfig)
+	faraday := newSubServer(&subServerCfg{
+		name:         FaradaySubServer,
+		remote:       g.cfg.faradayRemote,
+		remoteCfg:    g.cfg.Remote.Faraday,
+		macValidator: faradayServer,
+		macPath:      g.cfg.Faraday.MacaroonPath,
+		registerGrpcService: func(server grpc.ServiceRegistrar) {
+			frdrpc.RegisterFaradayServerServer(
+				server, faradayServer,
+			)
+		},
+		ownsURI: g.permsMgr.IsFaradayURI,
+		startIntegrated: func(_ lnrpc.LightningClient,
+			lndGrpc *lndclient.GrpcLndServices,
+			withMacaroonService bool) error {
+
+			return faradayServer.StartAsSubserver(
+				lndGrpc.LndServices, withMacaroonService,
+			)
+		},
+		stopIntegrated: faradayServer.Stop,
+		onStartSuccess: func() {
+			g.statusServer.setServerRunning(FaradaySubServer)
+		},
+		onStartError: func(err error) {
+			g.statusServer.setServerErrored(
+				FaradaySubServer, err.Error(),
+			)
+		},
+		onStop: func() {
+			g.statusServer.setServerStopped(FaradaySubServer)
+		},
+	})
+	g.subServerMgr.AddServer(faraday)
+
+	loopServer := loopd.New(g.cfg.Loop, nil)
+	loopd := newSubServer(&subServerCfg{
+		name:         LoopSubServer,
+		remote:       g.cfg.loopRemote,
+		remoteCfg:    g.cfg.Remote.Loop,
+		macPath:      g.cfg.Loop.MacaroonPath,
+		macValidator: loopServer,
+		ownsURI:      g.permsMgr.IsLoopURI,
+		registerGrpcService: func(server grpc.ServiceRegistrar) {
+			looprpc.RegisterSwapClientServer(server, loopServer)
+		},
+		startIntegrated: func(_ lnrpc.LightningClient,
+			lndGrpc *lndclient.GrpcLndServices,
+			withMacaroonService bool) error {
+
+			return loopServer.StartAsSubserver(
+				lndGrpc, withMacaroonService,
+			)
+		},
+		stopIntegrated: func() error {
+			loopServer.Stop()
+			return nil
+		},
+		onStartSuccess: func() {
+			g.statusServer.setServerRunning(LoopSubServer)
+		},
+		serverErrChan: loopServer.ErrChan,
+		onStartError: func(err error) {
+			g.statusServer.setServerErrored(
+				LoopSubServer, err.Error(),
+			)
+		},
+		onStop: func() {
+			g.statusServer.setServerStopped(LoopSubServer)
+		},
+	})
+	g.subServerMgr.AddServer(loopd)
+
+	poolServer := pool.NewServer(g.cfg.Pool)
+	poold := newSubServer(&subServerCfg{
+		name:         PoolSubServer,
+		remote:       g.cfg.poolRemote,
+		remoteCfg:    g.cfg.Remote.Pool,
+		macPath:      g.cfg.Pool.MacaroonPath,
+		macValidator: poolServer,
+		ownsURI:      g.permsMgr.IsPoolURI,
+		registerGrpcService: func(server grpc.ServiceRegistrar) {
+			poolrpc.RegisterTraderServer(server, poolServer)
+		},
+		startIntegrated: func(lndClient lnrpc.LightningClient,
+			lndGrpc *lndclient.GrpcLndServices,
+			withMacaroonService bool) error {
+
+			return poolServer.StartAsSubserver(
```

Comment:
Re that comment re converting many of the closures to be an interface, given this is just a pass thru call (same args and return value), a wrapper struct that embeds all the main daemon configs would take care of this nicely.

Reply:
turned out nicely I think